### PR TITLE
rgw: civetweb fixes for v1.1 upgrade

### PR DESCRIPTION
**Pull request by theanalyst**
- [x]  Update civetweb sha1 
- [x]  Resolve "validate_http_method" based on discussions in https://github.com/ceph/civetweb/pull/24
(update: this was readded as swift tests depend on this)
This changeset does the following:
- mg_request_info::mg_header is no longer a type (it is just mg_header, used auto instead)
- request_info->uri is deprecated, move to request_info->local_uri
- tentatively dropping `validate_http_method`, needs discussion

### CODE REVIEW

File: src/rgw/rgw_civetweb_frontend.cc [hunk 1]
```diff
@@ -55,7 +55,7 @@ int RGWCivetWebFrontend::run()
                    std::to_string(g_conf->rgw_thread_pool_size));
   set_conf_default(conf_map, "decode_url", "no");
   set_conf_default(conf_map, "enable_keep_alive", "yes");
-  set_conf_default(conf_map, "validate_http_method", "no");
+  //set_conf_default(conf_map, "validate_http_method", "no");
```
Comment (theanalyst) [inline src/rgw/rgw_civetweb_frontend.cc]: TODO: drop this once we decide whether it is necessary

**Comment by theanalyst** [comment]
Initial qe run: http://pulpito.ceph.com/abhi-2018-04-04_15:05:26-rgw-wip-civetweb-1.1-distro-basic-smithi/

We have failures in s3-tests with rfc2616 and swift failures (due to pip?), dropping the rfc2616 checks and rerunning as http://pulpito.ceph.com/abhi-2018-04-05_17:09:22-rgw-wip-civetweb-1.1-distro-basic-smithi/

**Comment by theanalyst** [comment]
I have updated the https://github.com/ceph/civetweb/pull/24 to correct branches. This PR needs to reference the correct sha1 once that merges

**Comment by theanalyst** [comment]
jenkins test this please

**Comment by theanalyst** [comment]
jenkins test this please (submodule sha not found as the remote didn't exist in ceph/civetewb)

**Comment by theanalyst** [comment]
@cbodley @mattbenjamin I have pushed the last changeset based on version CMake tells us, should we do it this way or go the objdump route?

**Comment by cbodley** [comment]
> @cbodley @mattbenjamin I have pushed the last changeset based on version CMake tells us, should we do it this way or go the objdump route?

i think the cmake route is fine for mimic. let's revisit the linking strategy with @mdw-at-linuxbox later

**Comment by theanalyst** [comment]
err that didn't work, version_greater_equal was introduced in cmake 3.7 only, I'll rewrite to do a greater than or equal to myself

**Comment by theanalyst** [comment]
The submodule change probably requires @mattbenjamin / @yehudasa  approval

**Pull request merged**